pkg/controlplane/control/manager.go

```diff
@@ -144,6 +144,99 @@ type Manager struct {
 	logger *logrus.Entry
 }
 
+// Restart coredns pods
+func coreDnsRestart(ctx context.Context, m *Manager) error {
+	var pods v1.PodList
+	if err := m.client.List(ctx, &pods, client.InNamespace("kube-system")); err != nil {
+		return err
+	}
+
+	for _, pod := range pods.Items {
+		if strings.Contains(pod.Name, "coredns") {
+			m.logger.Infof("Deleting pod: %s/%s.", pod.Namespace, pod.Name)
+			err := m.client.Delete(ctx, &pod)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// Add coredns rewrite for a given external dns service
+func addCoreDnsRewrite(ctx context.Context, m *Manager, name *types.NamespacedName, dnsName string) error {
+	corednsName := types.NamespacedName{
+		Name:      "coredns",
+		Namespace: "kube-system",
+	}
+	var cm v1.ConfigMap
+
+	if err := m.client.Get(ctx, corednsName, &cm); err != nil {
+		if k8serrors.IsNotFound(err) {
+			m.logger.Warnf("coredns configmap not found.")
+			return nil
+		} else {
+			return err
+		}
+	}
+	if data, ok := cm.Data["Corefile"]; ok {
+		// remove trailing end-of-line
+		dataEol := strings.TrimSuffix(data, "\n")
+		// break into lines
+		lines := strings.Split(dataEol, "\n")
+		serviceFqdn := fmt.Sprintf("%s.%s.svc.cluster.local", name.Name, name.Namespace)
+
+		var coreFileUpdated = false
+		var rewriteLine = ""
+		for i, line := range lines {
+			if strings.Contains(line, serviceFqdn) {
+				// matched line already exists
+				break
+			}
+			// ready marker is reached - matched line not found, append it here
+			if strings.Contains(line, "    ready") {
+				if strings.HasPrefix(dnsName, "*.") {
+					// wildcard dns
+					dnsName = strings.TrimPrefix(dnsName, "*")
+					dnsName = strings.ReplaceAll(dnsName, ".", "\\.")
+					dnsName = "(.*)" + dnsName
+
+					rewriteLine = fmt.Sprintf("    rewrite name regex %s %s answer auto", dnsName, serviceFqdn)
```

Comment:
nit: Is a tab expected before `rewrite` keyword? If so, consider using `\t` instead of spaces

Reply:
Yes, done.

Reply:
`\t` did not work although it is 4 chars. It might be that it is 4 spaces rather than tab. I reverted this back.